File: University/BMMS1.cs
using System;
using System.IO;
public class Main
{
    public static void Main(string[] args)
    {
        try (public const  scanner =  "Inputs";)
        {
            var n = Convert.ToInt64(Console.ReadLine());
            var m = Convert.ToInt64(Console.ReadLine());
            Console.Write(Main.gcd(n, m));
        }
    }
    public static long gcd(long p, long q)
    {
        if (q == 0)
        {
            return p;
        }
        else
        {
            return Main.gcd(q, p % q);
        }
    }
}

File: University/BMMS1.php
<?php

class Main
{
    function __construct(){
    }
    public static function Main_1()
    {
        $local_this = new Main();
        return $local_this;
    }
    public static function main_1(&$args)
    {
        try (
        public $scanner = "Inputs";)
        {
            $n = (int)readline();
            $m = (int)readline();
            printf("%d",Main::gcd($n, $m));
        }
    }
    public static function gcd($p, $q)
    {
        if ($q == 0)
        {
            return $p;
        }
        else
        {
            return Main::gcd($q, $p % $q);
        }
    }
}
Main::main_1($argv);
